### PR TITLE
 Change use of 'Frame' to 'Message' in XVIZ

### PR DESCRIPTION
**Pull request by twojtasz**
The term 'Frame' is overloaded, and in nearly all case
in XVIZ we mean an XVIZ Message.  We have created a distinction
where if we use 'Frame' it is not an individual message but the
set of data that constitutes a visual 'Frame' which could come from
multiple messages.  Therefore, 'Frame' is still used in @xviz/parse

This change includes the following:
- Changes to @xviz/builder, specifically getFrame() to getMessage()
- getFrame() is marked as deprecated in the source code and removed from docs

- @xviz/io changes all names using Frame to use Message
- @xviz/io shares json binary packing with @xviz/builder

- Documentation has been updated
- Examples conversions have been changed from convertFrame() to convertMessage()

- example/server has not been changed
- @xviz/parser has not been changed

### CODE REVIEW

File: docs/getting-started/mapping-vehicle-data.md [hunk 1]
```diff
@@ -38,8 +38,8 @@ important there is a required name `/vehicle_pose`. The list of
 [category values](/docs/protocol-schema/session-protocol.md#stream_metadata) can be seen in the
 specification.
 
-On every frame we must define the current Pose. We use the
-[XVIZBuilder](/docs/api-reference/xviz-builder.md#pose-streamid-) to create this for each frame.
+On every message we must define the current Pose. We use the
```
Comment (jlisee) [inline docs/getting-started/mapping-vehicle-data.md]: Do we really need the post for every message?

File: modules/builder/src/writers/xviz-writer/xviz-pack-binary.js [hunk 2]
```diff
@@ -68,6 +68,15 @@ export function packBinaryJson(json, gltfBuilder, objectKey = null, options = {}
 function flattenObject(key, object) {
   let typedArray = null;
   let size = 3;
+  // The overhead of creating binary data makes it inefficient unless the arrays
+  // are larger than an particular size.  From a byte-count perspective any point array
+  // with less than 20 entries results if increasing the overall size.  Therefore we
+  // only flatten entries with arrays > 20 elements.
+  const toTypedArrayMinimumLength = 20;
```
Comment (jlisee) [inline modules/builder/src/writers/xviz-writer/xviz-pack-binary.js]: Is this change supposed to be in here?

File: test/modules/builder/utils/flatten.spec.js [hunk 3]
```diff
@@ -67,3 +67,17 @@ test('flatten#flattenToTypedArray', t => {
   }
   t.end();
 });
+
+test('flattenToTypedArray#zero array returns null', t => {
```
Comment (jlisee) [inline test/modules/builder/utils/flatten.spec.js]: other change that snuck in here

File: docs/getting-started/mapping-vehicle-data.md (hunk 1 above)
Comment (twojtasz) [reply docs/getting-started/mapping-vehicle-data.md]: I changed this to add 'currently' as we plan to remove this requirement.
https://github.com/uber/xviz/issues/468

File: modules/builder/src/writers/xviz-writer/xviz-pack-binary.js (hunk 2 above)
Comment (twojtasz) [reply modules/builder/src/writers/xviz-writer/xviz-pack-binary.js]: Removed and will move to a new PR

File: test/modules/builder/utils/flatten.spec.js (hunk 3 above)
Comment (twojtasz) [reply test/modules/builder/utils/flatten.spec.js]: Removed and will move to a new PR

**Pull request merged**